client/gutenberg/extensions/presets/jetpack/editor.js

```diff
@@ -5,29 +5,22 @@
  */
 import './shared/public-path';
 import './editor-shared/block-category'; // Register the Jetpack category
-
-// TODO: Generate dyanmically from index.json
-// Appending `Block` to the names to keep `Map` from colliding with JS' Map
-import * as ContactFormBlock from 'gutenberg/extensions/contact-form';
-import * as MarkdownBlock from 'gutenberg/extensions/markdown';
-import * as MapBlock from 'gutenberg/extensions/map';
-import * as PublicizeBlock from 'gutenberg/extensions/publicize';
-import * as RelatedPostsBlock from 'gutenberg/extensions/related-posts';
-import * as SimplePaymentsBlock from 'gutenberg/extensions/simple-payments';
-import * as SubscriptionsBlock from 'gutenberg/extensions/subscriptions';
-import * as TiledGalleryBlock from 'gutenberg/extensions/tiled-gallery';
-import * as VRBlock from 'gutenberg/extensions/vr';
+import extensionSlugsJson from './index.json';
 import { isEnabled } from 'config';
 
-export default [
-	{ name: ContactFormBlock.name, settings: ContactFormBlock.settings },
-	...ContactFormBlock.fields,
-	MarkdownBlock,
-	MapBlock,
-	PublicizeBlock,
-	SimplePaymentsBlock,
-	SubscriptionsBlock,
-	...( isEnabled( 'jetpack/blocks/beta' )
-		? [ RelatedPostsBlock, TiledGalleryBlock, VRBlock ]
-		: [] ),
+const extensionSlugs = [
+	...extensionSlugsJson.production,
+	...( isEnabled( 'jetpack/blocks/beta' ) ? extensionSlugsJson.beta : [] ),
 ];
+
+export async function getExtensions() {
+	const promises = extensionSlugs.map( slug =>
+		import( '../../' + slug ).then( ( { children, name, settings } ) => ( {
```

Comment:
To reduce the extra imports and have webpack resolve and bundle in a single chunk, you should be able to set the mode:

https://webpack.js.org/api/module-methods/#magic-comments
```suggestion
		import(   /* webpackMode: "eager" */ '../../' + slug ).then( ( { children, name, settings } ) => ( {
```